### ISSUE TITLE
[Core Feature] Support DynamicWorkflow TaskResolver

### ISSUE THREAD

**Issue by EngHabu** [flyteorg/flyte#1177]
**Motivation: Why do you think this is important?**
I would like to be able to dictate which secrets to load based on inputs. The only way I could think of doing that is something like this:
```python
@dynamic
def my_task(secret_key: str) -> str:
   @task(secret_requests=[Secret(group="my-group", key=secret_key)])
   def inner_task() -> str:
      ...
   return inner_task()
```

**Describe alternatives you've considered**
1) Loading all keys for a secret. Doesn't work with some secret providers (particularly AWS Secret Manager)
2) This worked:
```python
@task(secret_requests=[Secret(group="my-group", key="")])
def inner_task() -> str:
    ...

@dynamic
def my_task(secret_key: str) -> str:
   inner_task.secret_requests = [Secret(group="my-group", key=secret_key)]

   return inner_task()
```

**Comment by kumare3** [comment]
@EngHabu after thinking a bit more, this seems to be not possible, or harder than it appears. This is because the internal structure can dynamically be altered based on inputs to the dynamic task. These inputs are not known at the time of resolving the inner task

**Comment by EngHabu** [comment]
I agree with that... I think the problem stands though... the solution can be different... or maybe the workaround I did is what we want to support (and document) ?
